pkg/sbproxy/reconcile/resyncer.go

```diff
@@ -58,3 +55,73 @@ func createResyncContext(ctx context.Context) (context.Context, string, error) {
 	entry := log.C(ctx).WithField(log.FieldCorrelationID, correlationID.String())
 	return log.ContextWithLogger(ctx, entry), correlationID.String(), nil
 }
+
+func (r *resyncJob) process(ctx context.Context) {
+	logger := log.C(ctx)
+	smBrokers, err := r.getSMBrokers(ctx)
+	if err != nil {
+		logger.Error(err)
+		return
+	}
+	smVisibilities, mappedPlans, err := r.getSMVisibilitiesAndPlans(ctx, smBrokers) // fetch as soon as possible
+	if err != nil {
+		logger.Error(err)
+		return
+	}
+
+	// get all the registered brokers from the platform
+	brokersFromPlatform, err := r.getBrokersFromPlatform(ctx)
+	if err != nil {
+		logger.WithError(err).Error("an error occurred while obtaining brokers from Platform")
+		return
+	}
+
+	r.reconcileBrokers(ctx, brokersFromPlatform, smBrokers)
+	r.reconcileVisibilities(ctx, smVisibilities, smBrokers, mappedPlans)
+}
+
+func (r *resyncJob) getSMBrokers(ctx context.Context) ([]platform.ServiceBroker, error) {
+	if r.platformClient.Broker() == nil {
+		return nil, errors.New("platform client cannot handle brokers. Cannot perform reconciliation")
```

Comment:
do we want to reconcile visibilities if we can't reconcile brokers? I assume not

Reply:
if reconciling one broker fails, we want to continue with the visibilities. If we cant get the brokers at all i would say we want to stop reconciling